amps/ags/rm-community/rm-community-repo/unit-test/java/org/alfresco/module/org_alfresco_module_rm/hold/HoldServiceImplUnitTest.java

```diff
@@ -306,6 +307,88 @@ public void setHoldReason()
         verify(mockedNodeService).setProperty(hold, PROP_HOLD_REASON, HOLD_REASON);
     }
 
+    @Test
+    public void setHoldDeletionReasonForNodeDoesNotExist()
+    {
+        // setup node service interactions
+        when(mockedNodeService.exists(hold))
+            .thenReturn(false)
+            .thenReturn(true)
+            .thenReturn(true);
+
+        // node does not exist
+        holdService.setHoldDeletionReason(hold, HOLD_DELETION_REASON);
+        verify(mockedNodeService, never()).setProperty(hold, PROP_HOLD_DELETION_REASON, HOLD_DELETION_REASON);
+    }
+
+    @Test
+    public void setHoldDeletionReasonForNodeIsNotAHold()
+    {
+        // setup node service interactions
+        when(mockedNodeService.exists(hold))
+            .thenReturn(true)
+            .thenReturn(true);
+
+        // node isn't a hold
+        holdService.setHoldDeletionReason(recordFolder, HOLD_DELETION_REASON);
+        verify(mockedNodeService, never()).setProperty(hold, PROP_HOLD_DELETION_REASON, HOLD_DELETION_REASON);
+    }
+
+    @Test
+    public void setHoldDeletionReason()
+    {
+        // setup node service interactions
+        when(mockedNodeService.exists(hold))
+            .thenReturn(true);
+
+        // set hold deletion reason
+        holdService.setHoldDeletionReason(hold, HOLD_DELETION_REASON);
+        verify(mockedNodeService).setProperty(hold, PROP_HOLD_DELETION_REASON, HOLD_DELETION_REASON);
+    }
+
+    @Test
+    public void updateHoldThatDoesNotExist()
+    {
+        // setup node service interactions
+        when(mockedNodeService.exists(hold))
+            .thenReturn(false);
+
+        // node does not exist
+        holdService.updateHold(hold, HOLD_NAME, HOLD_REASON, HOLD_DESCRIPTION);
+
+        verify(mockedNodeService, never()).setProperty(hold, ContentModel.PROP_NAME, HOLD_NAME);
+        verify(mockedNodeService, never()).setProperty(hold, ContentModel.PROP_DESCRIPTION, HOLD_DESCRIPTION);
+        verify(mockedNodeService, never()).setProperty(hold, PROP_HOLD_REASON, HOLD_DESCRIPTION);
```

Comment:
I think you can just check there are no interactions with the node service. Or maybe there are no calls to `setProperty`.

Reply:
Done.